ci/travis/ci.sh

```diff
@@ -38,21 +38,187 @@ should_run_job() {
   return "${skip}"
 }
 
+# Idempotent environment loading
 reload_env() {
   # TODO: We should really just use a new login shell instead of doing this manually.
   # Otherwise we might source a script that isn't idempotent (e.g. one that blindly prepends to PATH).
+
+  if [ -n "${GITHUB_PULL_REQUEST-}" ]; then
+    case "${GITHUB_PULL_REQUEST}" in
+      [1-9]*) TRAVIS_PULL_REQUEST="${GITHUB_PULL_REQUEST}";;
+      *) TRAVIS_PULL_REQUEST="false";;
+    esac
+    export TRAVIS_PULL_REQUEST
+  fi
+
+  local definitions=() paths=()
+
+  # Environment variables for all platforms
+  definitions+=(
+    PYTHON3_BIN_PATH=python
+    GOROOT="${HOME}/go"
+    GOPATH="${HOME}/go_dir"
+  )
+  # Environment variables for Windows
+  test "${OSTYPE}" != msys || definitions+=(
+    USE_CLANG_CL=1
+  )
+
+  # NOTE: Modifying PATH invalidates Bazel's cache! Do not add to PATH unnecessarily.
+  # PATH for all platforms
+  paths+=(
+    "${HOME}/miniconda/bin"
+  )
+  # PATH for Windows
+  test "${OSTYPE}" != msys || paths+=(
+    "${HOME}/miniconda/bin/Scripts"
+  )
+
+  export "${definitions[@]}" 2> /dev/null
+
+  prepend_PATH "${paths[@]}"
+}
+
+# Prepends each argument to PATH if it isn't already in the PATH.
+prepend_PATH() {
```

Comment:
try to get rid of this

Reply:
I'll look into it, but it may have repercussions and require a *lot* of refactoring to ensure we're spawning a new shell after environment modifications...